packages/backend/src/routers/dashboardRouter.ts

```diff
@@ -27,6 +32,36 @@ dashboardRouter.get(
     },
 );
 
+dashboardRouter.get(
+    '/availableFilters',
+    allowApiKeyAuthentication,
+    isAuthenticated,
+    async (req, res, next) => {
+        try {
+            const dashboard = await dashboardService.getById(
+                req.user!,
+                req.params.dashboardUuid,
+            );
+
+            const chartTiles = dashboard.tiles.filter(isDashboardChartTileType);
+            const savedQueryUuids = chartTiles
+                .map((tile) => tile.properties.savedChartUuid)
+                .filter((uuid): uuid is string => !!uuid);
```

Comment:
should I hide this logic inside dashboard service or is it fine?

also, are we updating openapi stuff or we don't care before we move to tsoa controllers?

Reply:
> should I hide this logic inside dashboard service or is it fine?

Yeah, I would have moved all of the logic in this router and `getAvailableFiltersForSavedQueries` to the dashboard service.